tests/e2e/prevent_duplicate_cluster_ids.go

```diff
@@ -0,0 +1,474 @@
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			for _, claim := range pvcs.Items {
+				pv := getPvFromClaim(client, namespace, claim.Name)
+				err := fpv.DeletePersistentVolumeClaim(client, claim.Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				ginkgo.By("Verify it's PV and corresponding volumes are deleted from CNS")
+				err = fpv.WaitForPersistentVolumeDeleted(client, pv.Name, poll,
+					pollTimeout)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				volumeHandle := pv.Spec.CSI.VolumeHandle
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(volumeHandle)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+					fmt.Sprintf("Volume: %s should not be present in the CNS after it is deleted from "+
+						"kubernetes", volumeHandle))
+			}
+		}()
+
+		ginkgo.By("Creating csi config secret with cluster id field set")
+		createCsiVsphereSecret(client, ctx, vCenterUIUser, vCenterUIPassword, csiNamespace,
+			vCenterIP, vCenterPort, dataCenter, "", "cluster1")
+
+		ginkgo.By("Restart CSI driver")
+		_, err = restartCSIDriver(ctx, client, csiNamespace, csiReplicas)
+		gomega.Expect(err).To(gomega.HaveOccurred())
+		ginkgo.By("Verify that cluster id configmap still exists")
+		verifyClusterIdConfigMapGeneration(client, ctx, csiNamespace, true)
+
+		ginkgo.By("Check if csi pods are in crashing state after recreation of secret with proper message")
+		csipods, err := client.CoreV1().Pods(csiNamespace).List(ctx, metav1.ListOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		k8sMasterIPs := getK8sMasterIPs(ctx, client)
+		k8sMasterIP := k8sMasterIPs[0]
+		var csiPodName string
+		for _, csiPod := range csipods.Items {
+			if strings.Contains(csiPod.Name, vSphereCSIControllerPodNamePrefix) {
+				csiPodName = csiPod.Name
+				break
+			}
+		}
+		errMessage := "Please remove the cluster ID from vSphere Config Secret."
+		grepCmdForErrMsg := "echo `kubectl logs " + csiPodName + " -n " +
+			csiSystemNamespace + " --allContainers" + " | grep " + "'" + errMessage
+
+		framework.Logf("Invoking command '%v' on host %v", grepCmdForErrMsg,
+			k8sMasterIP)
+		result, err := sshExec(sshClientConfig, k8sMasterIP,
+			grepCmdForErrMsg)
+		if err != nil || result.Code != 0 {
+			fssh.LogResult(result)
+			gomega.Expect(err).To(gomega.HaveOccurred(), fmt.Sprintf("couldn't execute command: %s on host: %v , error: %s",
+				grepCmdForErrMsg, k8sMasterIP, err))
+		}
+		if result.Stdout != "" {
+			framework.Logf("CSI pods are in crashing state with proper error message")
+		} else {
+			framework.Logf("CSI pods are in crashing state with improper error message")
+			gomega.Expect(err).To(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Remove cluster id field from vsphere config secret and verify csi pods are in running state")
+		recreateVsphereConfigSecret(client, ctx, vCenterUIUser, vCenterUIPassword, csiNamespace, vCenterIP,
```

Comment:
I didn't find defer block for reverting of config secret to its original value after testcase execution.

I can see that in AfterEach block but didn't find it here